1. Improvements in Base, System Reset and System Suspend service groups

pathakraul opened the pull request.

lftan commented on src/srvgrp-system-suspend.adoc:
```diff
@@ -9,132 +9,235 @@ ifndef::rootpath[]
+| 0
+| SUSPEND_TYPE
+| uint32
+| Suspend type
```
either link to line 32 or to Sleep types definitions in SBI spc.

pathakraul replied:
You mean this link https://github.com/riscv-non-isa/riscv-sbi-doc/blob/master/src/ext-sys-suspend.adoc#table_susp_sleep_types

lftan replied:
or link to Line 21 (not line 32) in this adoc. Can create a sub-section then reference link to it

lftan commented on src/srvgrp-system-suspend.adoc:
```diff
@@ -9,132 +9,235 @@ ifndef::rootpath[]
-| 0     | SUSPEND_TYPE	| uint32	| Suspend type value. +
-Refer to https://github.com/riscv-non-isa/riscv-sbi-doc/blob/master/src/ext-sys-suspend.adoc#table_susp_sleep_types[*SUSP System Sleep Types*^]
-in the RISC-V SBI Specification cite:[SBI] for the SUSPEND_TYPE definition.
+| Word 
```
in the new pdf,  font size of italic text is look bigger than other text. This can be changed?
![image](https://github.com/user-attachments/assets/5fed6187-4141-4d64-81a4-32f78f9feac3)

pathakraul replied:
We are relying on the format theme coming from the docs_resource submodule.

I think your repo has not the updated docs_resources folder. This is how it currently looks
![image](https://github.com/user-attachments/assets/ccbe0492-f338-4e42-b8c6-afb54dd60026)

lftan replied:
font size looks okay after update doc-resources

lftan commented on src/srvgrp-system-suspend.adoc:
```diff
@@ -9,132 +9,235 @@ ifndef::rootpath[]
+| 1     
+| SUSPEND_TYPE	
+| uint32	
+| Suspend type.
```
Same here, reference link to 'Suspend type'

lftan commented on src/srvgrp-system-suspend.adoc:
```diff
@@ -9,132 +9,235 @@ ifndef::rootpath[]
+! Error Code
+! Description
+
+! RPMI_ERROR_NOT_SUPPORTED
```
other services has RPMI_SUCCESS for success, need have this error code for all services?

lftan replied:
here also, make it one line.

![image](https://github.com/user-attachments/assets/b307e0ce-fdb9-4075-b567-f73b833dcf84)

lftan commented on src/srvgrp-system-reset.adoc:
```diff
@@ -9,128 +9,187 @@ ifndef::rootpath[]
+| 0     
+| RESET_TYPE	
+| uint32	
+| Reset type
```
add the reference link to line 30

lftan commented on src/srvgrp-system-suspend.adoc:
```diff
@@ -9,132 +9,235 @@ ifndef::rootpath[]
+! RPMI_ERROR_NOT_FOUND 
+! EVENT_ID is not supported or invalid.
+
+! RPMI_ERROR_NOT_SUPPORTED 
```
Adjust the table column width to fix the whole text in one line.

In SYSSUSP_ENABLE_NOTIFICATION table:
![image](https://github.com/user-attachments/assets/c623aaeb-39e1-4510-81ac-33a2c83888e0)

lftan commented on src/srvgrp-system-reset.adoc:
```diff
@@ -9,128 +9,187 @@ ifndef::rootpath[]
+! RPMI_ERROR_NOT_FOUND
+! `EVENT_ID` is not supported or invalid.
+
+! RPMI_ERROR_NOT_SUPPORTED
```
same here, make it display in one line.

pathakraul commented:
Addressed all the comments, hopefully without missing any. 
Also pushed another commit in this PR for a incorrect statement in message protocol chapter

pathakraul commented:
I pushed commit for Base service group also

lftan commented on src/srvgrp-system-suspend.adoc:
```diff
@@ -9,132 +9,244 @@ ifndef::rootpath[]
+been complete.
+
+==== Suspend Types
+RPMI supports suspend states and their values as defined by SBI
```
Use 'suspend types' instead of suspend states here?

lftan commented on src/srvgrp-system-suspend.adoc:
```diff
@@ -9,132 +9,244 @@ ifndef::rootpath[]
-| 0x01		| SYSSUSP_ENABLE_NOTIFICATION	| NORMAL_REQUEST
-| 0x02		| SYSSUSP_GET_ATTRIBUTES	| NORMAL_REQUEST
-| 0x03		| SYSSUSP_SUSPEND		| NORMAL_REQUEST
+| Service ID
```
This table should move to the "Suspend Types" sub-section. It is under chapter "4.3.1. Suspend Types" now.

lftan commented on src/srvgrp-system-reset.adoc:
```diff
@@ -9,128 +9,198 @@ ifndef::rootpath[]
-| 0x01		| SYSRST_ENABLE_NOTIFICATION	| NORMAL_REQUEST
-| 0x02		| SYSRST_GET_ATTRIBUTES		| NORMAL_REQUEST
-| 0x03		| SYSRST_RESET			| POSTED_REQUEST
+| Service ID
```
same here, move to before subsection of "Reset Types"

lftan commented on src/srvgrp-system-reset.adoc:
```diff
@@ -9,128 +9,198 @@ ifndef::rootpath[]
+| RESET_TYPE	
+| uint32	
+| Reset type +
+Refer <<Reset Types>> section for supported reset types.
```
can remove 'section'here, generated pdf has Section in front.
eg:
Refer Section 4.2.1 section for supported reset types.

lftan commented on src/srvgrp-system-reset.adoc:
```diff
@@ -9,128 +9,198 @@ ifndef::rootpath[]
+| RESET_TYPE
+| uint32
+| Reset type +
+Refer <<Reset Types>> section for supported reset types.
```
same here

lftan commented on src/srvgrp-base.adoc:
```diff
@@ -9,133 +9,142 @@ ifndef::rootpath[]
-A service group is a collection of services which are logically grouped
-according to the functionality they provide. For example, all voltage related
- services/messages are collectively referred to as the Voltage Service Group.
+=== Service Group: BASE (servicegroup_id: 0x0001)
```
In the Message Format chapter, we use "SERVICEGROUP_ID", should standardize this?
Same for "SERVICE_ID" in each sub-chapter header.

lftan commented on src/srvgrp-base.adoc:
```diff
@@ -9,133 +9,142 @@ ifndef::rootpath[]
+| Description
+
+| 0x001		
+| REQUEST_HANDLE_ERROR	
```
adjust table width, so that fix whole word in one line

lftan commented on src/srvgrp-base.adoc:
```diff
@@ -145,26 +154,48 @@ and Minor numbers.
+==== Service: BASE_GET_IMPLEMENTATION_ID (service_id: 0x03)
+This service is used to get a `32-bit` RPMI implementation ID assigned to the
+software which implements the RPMI specification. Every implementation ID is
+unique and listed in the table <<table_base_rpmi_impl_id>>
```
It display like this, fix it.

![image](https://github.com/user-attachments/assets/f43d3504-6668-4f84-b433-397cc1f96caf)

lftan commented on src/srvgrp-base.adoc:
```diff
@@ -224,75 +293,99 @@ RPMI implementation on the PuC. Each vendor will be assigned a unique Vendor ID.
+| Platform Identifier field length in bytes.
+
+| 2
+| PLATFORM_ID
```
make them display in one line
![image](https://github.com/user-attachments/assets/c44b7470-8217-4ba9-957b-d92aec90f131)

lftan commented on src/srvgrp-base.adoc:
```diff
@@ -224,75 +293,99 @@ RPMI implementation on the PuC. Each vendor will be assigned a unique Vendor ID.
-	0: Service group not implemented by platform.
-	1: Service group implemented by platform.
+| 1
+| SERVICE_GROUP_STATUS
```
make this display in one line

pathakraul commented:
Merging this since other PRs are dependent on this, we have more review stages coming in
